VideoApi/VideoApi/Services/VirtualRoomService.cs

```diff
@@ -42,43 +43,80 @@ public async Task<Room> GetOrCreateAnInterpreterVirtualRoom(Conference conferenc
                 .DefaultIfEmpty()
                 .Max();
             var room = GetRoomForParticipant(conferenceId, interpreterRooms, participant) ??
-                       await CreateAnInterpreterRoom(conference, count);
+                       await CreateAnInterpreterRoom(conference, count, roomPrefix);
 
             return room;
         }
-
-        private async Task<Room> CreateAnInterpreterRoom(Conference conference, int existingRooms)
+        
+        private async Task<Room> CreateAnInterpreterRoom(Conference conference, int existingRooms, string roomPrefix)
         {
             _logger.LogInformation("Creating a new interpreter room for conference {Conference}", conference.Id);
-            var createRoomCommand = new CreateRoomCommand(conference.Id, null, VirtualCourtRoomType.Civilian, false);
-            await _commandHandler.Handle(createRoomCommand);
-            var roomId = createRoomCommand.NewRoomId;
+            var roomId = await CreateRoom(conference.Id, VirtualCourtRoomType.Civilian);
             _logger.LogInformation("Creating a new civilian room {Room} for conference {Conference}", roomId,
                 conference.Id);
             var ingestUrl = $"{conference.IngestUrl}/{roomId}";
-            var roomPrefix = "Interpreter";
 
+            var vmr = await CreateVmr(conference, roomId, roomPrefix, ingestUrl, VirtualCourtRoomType.Civilian, "I",
+                existingRooms);
+
+            await UpdateRoomConnectionDetails(conference, roomId, vmr, ingestUrl);
+            _logger.LogDebug("Updated room {Room} for conference {Conference} with joining details", roomId,
+                conference.Id);
+            return await GetUpdatedRoom(conference, roomId);
+        }
+
+        public async Task<Room> GetOrCreateAWitnessVirtualRoom(Conference conference, Participant participant)
+        {
+            var conferenceId = conference.Id;
+            var roomPrefix = "Witness";
```

Comment:
@shaed-parkar  room prefix should still be Interpreter. These are still interpreter rooms and do not change based on the hearing role role of the participant.

Reply:
Only partcipant_type we send to Kinly at create room stage change here.